x-pack/plugins/apm/public/components/app/ServiceOverview/view.js

```diff
@@ -35,8 +35,9 @@ class ServiceOverview extends Component {
     this.checkForHistoricalData(this.props);
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.checkForHistoricalData(nextProps);
+  componentDidUpdate() {
+    // QUESTION: Do we want to check on ANY update, or only if serviceList status/data have changed?
+    this.checkForHistoricalData(this.props);
```

Comment:
Inside `checkForHistoricalData` we already have a check, so we don't fetch whenever the component updates. The check is currently:
```js
const shouldFetch = serviceList.status === STATUS.SUCCESS && isEmpty(serviceList.data)
```

Could probably make it a little stricter with:
```js
const shouldFetch = serviceList.status === STATUS.SUCCESS && isEmpty(serviceList.data) && this.state.historicalDataFound
```

Reply:
Yeah that makes sense. Just for conversation, here's a little more on what I was thinking here:

* If this component were connected to a react-router `<Route>` or if it eventually was given another prop etc etc, anytime any other prop changes, this function will get called.
* Any time this function is called, if the conditions are still met, we'll re-check for historical data
* So in theory, we could be re-checking because the location state is updating, or a new unrelated prop is changing, etc.

In this case it might not be a problem because we know what props are coming in, etc., but sometimes I like to wrap calls in `cDU` with something like:
```js
if (prevProps.serviceList !== this.props.serviceList) {
  // do check
}
```
Think that's overkill for this situation?

Reply:
Definitely not a big deal to me here so I'm gonna merge this, we can talk about this kind of check as a general rule over time if we think it makes sense.

Reply:
Thanks for merging, and let's re-evaluate how components update next time it comes up.